mipengine/controller/node_landscape_aggregator.py

```
@@ -23,107 +22,80 @@
+        try:
+            tmp = tasks_handler.queue_node_info_task(
+                request_id=NODE_LANDSCAPE_AGGREGATOR_REQUEST_ID
+            )
+            async_results[tasks_handler] = tmp
+        except (CeleryConnectionError, CeleryTaskTimeoutException) as exc:
+            # just log the exception do not reraise it
+            logger.warning(exc)
+
+    nodes_info = []
+    for tasks_handler, async_result in async_results.items():
+        try:
+            result = tasks_handler.result_node_info_task(async_result=async_result)
+            nodes_info.append(result)
+        except (CeleryConnectionError, CeleryTaskTimeoutException) as exc:
+            # just log the exception do not reraise it
+            logger.warning(exc)
 
     return nodes_info
 
 
-async def _get_node_datasets_per_data_model(
+def _get_node_datasets_per_data_model(
     node_socket_addr: str,
 ) -> Dict[str, Dict[str, str]]:
-    celery_app = get_node_celery_app(node_socket_addr)
-    task_signature = celery_app.signature(GET_NODE_DATASETS_PER_DATA_MODEL_SIGNATURE)
+    tasks_handler = NodeInfoTasksHandler(
+        node_queue_addr=node_socket_addr, tasks_timeout=CELERY_TASKS_TIMEOUT
+    )
 
     try:
-        datasets_per_data_model = await _task_to_async(task_signature, app=celery_app)(
+        async_result = tasks_handler.queue_node_datasets_per_data_model_task(
             request_id=NODE_LANDSCAPE_AGGREGATOR_REQUEST_ID
         )
-    except Exception as exc:
-        logger.error(
-            f"Error at 'get_node_datasets_per_data_model' in node '{node_socket_addr}': {type(exc)}:{exc}"
+        datasets_per_data_model = (
+            tasks_handler.result_node_datasets_per_data_model_task(
+                async_result=async_result
+            )
         )
-        datasets_per_data_model = {}
-
-    return datasets_per_data_model
+        return datasets_per_data_model
+    except (CeleryConnectionError, CeleryTaskTimeoutException) as exc:
```

All Exceptions should be caught here and an empty dict should be returned, as in the previous implementation.

If an exception occurs here, we don't want to fail the NLA, just consider that his node doesn't have datasets.

---

if it gets in the `except ..` block there is no return statement so when the function returns will return a `NoneType` which is more indicative that something went wrong than returning an empty dictionary.
"just consider that his node doesn't have datasets." why would a node not having datasets would cause an exception to be raised?

---

The point of returning an empty dict instead of a `NoneType` was to contain the logic of getting the datasets only inside this class. If we return `None` we need to also have some part of that logic outside. 

What I meant with "just consider that his node doesn't have datasets." was that we need to catch all exceptions, not just specific ones.

---

ok, re-added returning empty dictionary if exception is caught in `_get_node_datasets_per_data_model`